tin_lookup.go

```diff
@@ -0,0 +1,13 @@
+package gobltin
+
+// Interface for all lookup functions
+type CheckTinResponse struct {
+	Valid       bool   `json:"valid"`
+	CountryCode string `json:"countryCode"`
+	TinNumber   string `json:"vatNumber"`
+	RequestDate string `json:"requestDate"`
+}
+
+type TinLookup interface {
+	LookupTin(countryCode, tinNumber string) (*CheckTinResponse, error)
```

Comment:
You want to include a "context" here, which is an unusual but highly recommended convention in Go. I'd also recommend just passing in the GOBL `tax.Identity` object:

```suggestion
	LookupTin(ctx context.Context, tid *tax.Identity) (*CheckTinResponse, error)
```